ForkMe-Mobile/app/src/main/java/dlei/forkme/gui/activities/IntermediateActivity.java

```diff
@@ -0,0 +1,159 @@
+        okHttpBuilder.addInterceptor(new Interceptor() {
+            @Override
+            public okhttp3.Response intercept(Chain chain) throws IOException {
+                // Manipulate request to add headers.
+                // Can't mutate the request but can make a new one.
+                Request request = chain.request();
+                Request.Builder newRequest = request.newBuilder()
+                        // Add in user access token.
+                        .addHeader("Authorization", "token " + AppSettings.sOAuthToken);
+                // Pass on our request to execute.
+                return chain.proceed(newRequest.build());
+            }
+        });
+
+        Retrofit retrofit = new Retrofit.Builder()
+                .client(okHttpBuilder.build())
+                .baseUrl(BaseUrls.githubApi)
+                .addConverterFactory(GsonConverterFactory.create())
+                .build();
+        GithubApi endpoint = retrofit.create(GithubApi.class);
+        Call<User> call = endpoint.getAuthenticatedUser();
+
+        call.enqueue(new Callback<User>() {
+            @Override
+            public void onResponse(Call<User> call, Response<User> response) {
+                if (response.code() == 200 && response.isSuccessful()) {
+                    User u = response.body();
+                    // Initalize user details.
+                    AppSettings.setUserLogin(u.getLogin());
+                    AppSettings.setUserName(u.getName());
+                    AppSettings.setUserAvatarUrl(u.getAvatarUrl());
+                    AppSettings.setUserEmail(u.getEmail());
+
+                    // Get information from db.
+                    mDbHelper = DatabaseHelper.getDbInstance(getApplicationContext());
+                    try {
+                        // Loads settings from persistent storage to AppSettings attributes.
+                        mDbHelper.loadSettings();
+                        Log.d("IntermediateActivity: ", "Data loaded from db");
+                    } catch (NoDataException e) {
+                        Log.d("No data: ", "No data from db");
+                        if (!AppSettings.sUserLogin.equals("")) {
+                            mDbHelper.insertSettings();
+                        } else {
+                            // Should never happen.
+                            Log.wtf("IntermediateActivity: ", "AppSettings.sUserLogin has no value");
+                        }
+                    } catch (TooMuchDataException e) {
+                        // Should not ever happen.
+                        Log.wtf("IntermediateActivity: ", "Too much data from db");
+                    }
+                    // Up to here: AppSettings will always have most up to date data.
+
+
+                    Log.d("IntermediateActivity: ", "getAuthenticatedUser: " + u.getLogin());
+                    Intent intent = new Intent(getApplicationContext(), TrendingRepositoriesActivity.class);
```

Comment:
Swap out getApplicationContext for this in Intent constructors inside of Activity objects.